ext/src/dll/opentelemetry_cpp.src

```diff
@@ -0,0 +1,18 @@
+// clang-format off
+LIBRARY opentelemetry_cpp
+EXPORTS
+    ?Create@OStreamSpanExporterFactory@trace@exporter@v1@opentelemetry@@SA?AV?$unique_ptr@VSpanExporter@trace@sdk@v1@opentelemetry@@U?$default_delete@VSpanExporter@trace@sdk@v1@opentelemetry@@@std@@@std@@XZ
```

Comment:
This file is checked in in git, so it is considered source code.

I assume it was generated somehow.

For unix users sending PR to the opentelemetry-cpp repository, how are users without access to a windows machine expected to come up with the mangled symbol syntax used here for exports ?

Reply:
This is manually edited as these are just decorated name. I was considering adding the undecorated symbol name to this file for better readability.

This file is dedicated for Windows. Linux related PR should not need to touch this unless changing API/Factory method which should be very rare? But the fallback is the build break message will show the unresolved decorated symbol(s) which might be added here.

Reply:
Thanks for the clarifications.

Please add the undecorated names in comments, it will help to search the source code (for example, when searching for "OStreamSpanExporterFactory::Create").

Changing exported apis should be rare, but adding new ones will be more frequent.

Reply:
Ok. Added the undecorated names to the comment.

Reply:
Is there any document about how to generate this file or can we write some script to generate it automatically?It will be very helpful for developers who is not familiar with this(Such as me).The experimental APIs may still changes frequently.

Reply:
I am not sure whether there is a easy way to automatically generate this. Currently I am doing is extracting the unresolved decorated symbol from the error message as below, and check the undecorated symbol (also in the error message below) and decide whether it makes sense to export this symbol, and add it to the list finally. For example, the unresolved symbol below is https://github.com/open-telemetry/opentelemetry-cpp/blob/main/sdk/src/logs/batch_log_record_processor_factory.cc#L15. I tweaked the http_log example to call `BatchLogRecordProcessorFactory::Create`. Hope this helps.

> http_log_main.cc.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: static class std::unique_ptr<class opentelemetry::v1::sdk::logs::LogRecordProcessor,struct std::default_delete<class opentelemetry::v1::sdk::logs::LogRecordProcessor> > __cdecl opentelemetry::v1::sdk::logs::BatchLogRecordProcessorFactory::Create(class std::unique_ptr<class opentelemetry::v1::sdk::logs::LogRecordExporter,struct std::default_delete<class opentelemetry::v1::sdk::logs::LogRecordExporter> > &&,struct opentelemetry::v1::sdk::logs::BatchLogRecordProcessorOptions const &)" (__imp_?Create@BatchLogRecordProcessorFactory@logs@sdk@v1@opentelemetry@@SA?AV?$unique_ptr@VLogRecordProcessor@logs@sdk@v1@opentelemetry@@U?$default_delete@VLogRecordProcessor@logs@sdk@v1@opentelemetry@@@std@@@std@@$$QEAV?$unique_ptr@VLogRecordExporter@logs@sdk@v1@opentelemetry@@U?$default_delete@VLogRecordExporter@logs@sdk@v1@opentelemetry@@@std@@@7@AEBUBatchLogRecordProcessorOptions@2345@@Z) referenced in function "void __cdecl `anonymous namespace'::InitLogger(void)" (?InitLogger@?A0x55fc834a@@YAXXZ)

Reply:
@ThomsonTan Can we add some comment in this file to explain
  - when to add symbols in this file.
  - how to get these decorated symbols e.g., if there is tool (say objdump/ dependency walker) to get the mangled symbols from object file, and put it here. 
  
Not blocker, we can add these a separate PR, but would be helpful for other developers.